packages/docusaurus-plugin-content-docs/src/options.ts

```diff
@@ -70,6 +71,7 @@ const VersionsOptionsSchema = Joi.object()
 
 const OptionsSchema = Joi.object<PluginOptions>({
   path: Joi.string().default(DEFAULT_OPTIONS.path),
+  versionedDocsPath: Joi.string(),
```

Comment:
If the default value is the site dir, I would find a way to make Joi (or at least the option normalization process to set this value to the site dir directly. This way we don't have to do `path.join(siteDir, versionedDocsPath  || '')` later in the process and can rely on a reliable and normalized option value that really describes a path